src/net/echo.zig

```diff
@@ -259,10 +259,10 @@ pub fn handleRequest(
 
         // convert a u32 to Ipv4
         var socket_addr = SocketAddr.initIpv4(.{
-            @as(u8, @intCast(addr.sa.addr >> 24 & 0xFF)),
-            @as(u8, @intCast(addr.sa.addr >> 16 & 0xFF)),
-            @as(u8, @intCast(addr.sa.addr >> 8 & 0xFF)),
             @as(u8, @intCast(addr.sa.addr & 0xFF)),
```

Comment:
I wonder how the test below has been working with this reversed.

It could be good to extract out this logic into a conversion function for Address->SocketAddr, and write a test for it.

Reply:
Well it wasn't actually testing the ip address being that in order to assert the ip, you'd need to know the ip which is a circular dependency haha

Reply:
On second thought, I can at the very least assert localhost ip addr - done